src/Workspaces/Remote/ServiceHub/Host/RemoteAnalyzerAssemblyLoaderService.cs

```diff
@@ -22,9 +22,19 @@ internal sealed class RemoteAnalyzerAssemblyLoaderService(
     [ImportMany] IEnumerable<IAnalyzerAssemblyResolver> externalResolvers)
     : IAnalyzerAssemblyLoaderProvider
 {
-    private readonly ShadowCopyAnalyzerAssemblyLoader _shadowCopyLoader =
-        new(AbstractAnalyzerAssemblyLoaderProvider.GetDefaultShadowCopyPath(), externalResolvers.ToImmutableArray());
+    /// <summary>
+    /// Default shared instance, for all callers who do not want to provide a custom AssemblyLoadContext.
+    /// </summary>
+    public IAnalyzerAssemblyLoaderInternal SharedShadowCopyLoader { get; } = CreateLoader(externalResolvers);
 
-    public IAnalyzerAssemblyLoaderInternal GetShadowCopyLoader()
-        => _shadowCopyLoader;
+    private static ShadowCopyAnalyzerAssemblyLoader CreateLoader(IEnumerable<IAnalyzerAssemblyResolver> externalResolvers)
+        // Note: we use a different path here than what is used on the host.  First, using the same path wouldn't
+        // actually provide any benefits, as the shadow copy system already always makes a unique guid-based directory
+        // under the base directory passed in.  Second, every fresh loader attempts to delete the other directories off
+        // of this base as a cleanup pass.  We don't want the host or the remote side cleaning up the other as each
+        // already handles that, and this just ends up with more IO exceptions trying to cleanup locked directories.
```

Comment:
Random aside: we used to have a method where the process held a Mutex with the GUID so other processes could see if that folder was still in use rather than trying and failing with exceptions. Are we not doing that?

Reply:
not sure where we've landed with this.  if this is the case, i think we could jsut have a single location for all analyzer shadow copying :)